### PR TITLE
Fixes to menus and UI

### LINKED ISSUE #8: While in-game, settings menu UI elements scale incorrectly on high resolutions
**Describe the bug**
While in a level, If the game window is large enough to the point that the UI elements are scaled up to compensate, most elements in the settings menu appear oversized, becoming difficult to use. Only exceptions are the "Show Timer" text, and the language selection.

**To Reproduce**
1. Enter any level
2. Press P to pause, and go to the Settings menu
3. Increase the size of the window, go fullscreen if needed
4. Observe the options becoming oversized.

**Expected behavior**
UI elements in the settings menu while in-game, should work as they do in the main menu, remaining reasonably sized.

**Screenshots**
image
For comparison -- in the main menu, UI scales appropriately:
image

**System Information**
 - OS/Browser: Linux/Windows (any platform with a resizable window and high-res displays)
 - Game version: v0.2.0.alpha

### PR DESCRIPTION
**Pull request by GTcreyon**
# Description of changes
- Fix scaling in options menu for both the main menu and the pause menu
- Rework scaling of the HUD for different aspect ratios
- Improve how margins of HUD elements are handled
- Rework the level info menu to use a VBoxContainer
- Fix one-pixel gaps at the edge of the screen
- Refactor various scripts, notably the menu buttons on the pause menu
- Remove redundant code

# Reviewers
@jaschutte @Koopa1018

# Issue(s)
Closes #8

### CODE REVIEW

File: scenes/menus/title/main_menu/main_menu.gd [hunk 1]
```diff
@@ -198,27 +198,8 @@ func _cycle_through(direction: int) -> void:
...
-	border.margin_bottom = OS.window_size.y / scale
-	
-	options_menu.margin_top = 26 * scale
-	options_menu.margin_bottom = -4 * scale
-	options_menu.margin_left = 4 * scale
-	options_menu.margin_right = -4 * scale
-	
-	back_button.rect_scale = Vector2.ONE * scale
-	back_button.rect_pivot_offset.x = back_button.rect_size.x
+	rect_scale = Vector2.ONE * scale
```
Comment (jaschutte) [inline scenes/menus/title/main_menu/main_menu.gd:201]: Why Vector2.ONE * scale?

Comment (GTcreyon) [reply scenes/menus/title/main_menu/main_menu.gd:201]: Because Vector2(scale, scale) is inconvenient, imo.

File: shaders/glow.shader [hunk 2]
```diff
@@ -13,6 +14,10 @@ void vertex()
 	// increase the size by the the width, this extra space is needed for the outline
 	extra_movement = TEXTURE_PIXEL_SIZE * width * 2.;
 	VERTEX *= 1. + extra_movement;
+	if (recenter)
+	{
+		VERTEX -= vec2(1, 1) * width;
```
Comment (jaschutte) [inline shaders/glow.shader:19]: I'm pretty sure the syntax `vec2(width)` works in glsl

**Review by jaschutte — APPROVED** [review]

File: shaders/glow.shader (hunk 2 above)
Comment (GTcreyon) [reply shaders/glow.shader:19]: True, I'll fix that

**Comment by GTcreyon** [comment]
Some notes:
- The Map menu still isn't done, and the scrolling doesn't work correctly
- The FLUDD menu doesn't exist, but that can be worked on now
- The Options menu doesn't scroll yet, but that'll be a separate PR
- The Exit menu still doesn't exist (see #43), but that can be worked on now

File: scenes/menus/title/main_menu/main_menu.gd (hunk 1 above)
Comment (jaschutte) [reply scenes/menus/title/main_menu/main_menu.gd:201]: Oops, I wasn't aware scale was a number, I thought it was a vector2, nvm then

**Pull request merged**